api/src/main/java/org/apache/any23/configuration/Setting.java

```diff
@@ -113,6 +113,8 @@ public final Type getValueType() {
     }
 
     /**
+     * @param setting a setting key to retrieve
+     * @param <S> generic type for setting
```

Comment:
Perhaps something more along the lines of:

```
@param setting a setting that may or may not have the same key as this setting
@param <S> the type of the supplied setting
```

Reply:
Thanks